src/query-helpers.js

```diff
@@ -84,16 +89,70 @@ function queryAllByProp(
     .map(removeBadProperties);
 }
 
-function queryByProp(...args) {
-  return firstResultOrNull(queryAllByProp, ...args);
+function queryByProp(prop, container, match, ...args) {
+  const els = queryAllByProp(prop, container, match, ...args);
+  if (els.length > 1) {
+    throw getMultipleElementsFoundError(
+      `Found multiple elements by [${prop}=${getPropMatchAsString(match)}]`,
+      container,
+    );
+  }
+  return els[0] || null;
+}
+
+// this accepts a query function and returns a function which throws an error
+// if more than one elements is returned, otherwise it returns the first
```

Comment:
super nitpicky, but I'd do the line breaks differently for this comment. Currently it's easy to read this as it always throws an error